### PR TITLE
Using `string constants` for `error messages`

### LINKED ISSUE #208: Using String constants for error messages
**Description**
Consider using named string constants to represent error messages, suggestions, and other static text in the codebase. This practice offers advantages in terms of maintainability and consistency.

**Motivation**
- Centralized Management: Defining messages as constants at the top of the code or in a dedicated file makes it easier to manage and update them consistently.

- Consistency: Using constants ensures uniformity in error messages and suggestions, improving clarity and user experience.

- Readability: Named constants enhance code readability by providing clear context for the purpose of each message.

**Additional context**
https://medium.com/codex/how-to-share-constants-in-typescript-project-8f76a2e40352#:~:text=It%20is%20a%20best%20practice,maintainability%2C%20and%20enforce%20type%20safety.

### PR DESCRIPTION
**Pull request by Sadaf-A**
This PR adds string constants for errors.
Resolves #208

### CODE REVIEW

**Comment by Sadaf-A** [comment]
I believe same can be done for suggestion messages

**Review by RonConfigu — COMMENTED** [review]
Hi @Sadaf-A,

Thank you for opening the issue and this PR. 💯

This issue has been lingering in the back of our minds for a while now, and seeing the community address it definitely elevates its importance.

That being said, we will need to devise a comprehensive solution for this and establish its infrastructure before these changes can be applied.

We will get in touch with you once such a plan is in place, and we would be delighted to accept your contribution.

Thanks, Ron

File: ts/packages/ts/src/ErrorConstants.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export const INVALID_PATH_ERROR = 'invalid path "%s"';
+export const INVALID_KEY_ERROR = 'invalid key "%s"';
+export const INVALID_TYPE_PROPERTY_ERROR = 'invalid type property';
+export const INVALID_DEFAULT_PROPERTY_ERROR = 'invalid default property';
+export const INVALID_DEPENDS_PROPERTY_ERROR = 'invalid depends property';
+
+
+
```
Comment (jamesL92) [inline ts/packages/ts/src/ErrorConstants.ts:8]: A few more trailing newlines than needed :sweat_smile:

**Review by jamesL92 — CHANGES_REQUESTED** [review]

**Comment by rannn505** [comment]
@Sadaf-A, thank you for your input. As @RonConfigu pointed out, we're planning a comprehensive consolidation of our error messages across the entire project. I'll keep you updated on when we kickstart this initiative, and we'd welcome your contribution on the respective PR.

**Pull request closed without merging**